nilearn/input_data/nifti_spheres_masker.py

```diff
@@ -138,6 +220,51 @@ def __call__(self, imgs):
         return signals, None
 
 
+class _InversionFunctor(object):
+
+    func_name = 'nifti_spheres_masker_inversion'
+
+    def __init__(self, seeds_, radius, mask_img, allow_overlap, dtype):
+        self.seeds_ = seeds_
+        self.radius = radius
+        self.mask_img = mask_img
+        self.allow_overlap = allow_overlap
+        self.dtype = dtype
+        # Mask is only used to get affine and shape
+        if self.mask_img is not None:
+            mask = check_niimg_3d(self.mask_img)
+            self.target_affine = mask.affine
+            self.target_shape = mask.shape
+        else:
```

Comment:
Sorry, but this is black magic: we 're figuring out an affine while there is None. I think we should rather spit an error here (with a clear message) ?

Reply:
Great for me. This was the big trouble I had with the `.inverse_transform()` in the first place.